### ISSUE TITLE
Don't use `join_all` on `JoinHandle`

### ISSUE THREAD

**Issue by Darksonn** [rust-lang/rust-clippy#9843]
### What it does

The `join_all` method is used to run multiple futures concurrently by polling all of them simultaneously. However, if used together with `tokio::spawn`, it provides no advantage over a simple loop, and has a performance impact since the `join_all` does extra work to poll all of them simultaneously.

### Lint Name

join-all-joinhandles

### Category

perf

### Advantage

 - Using `join_all` on a vector of `JoinHandle` is inefficient.
 - Using `join_all` makes it easy to ignore errors because `Vec<Result<...>>` is not `#[must_use]`.

### Drawbacks

Calling `join_all` is shorter.

### Example

```rust
let join_handles: Vec<JoinHandle<T>> = ...;
join_all(join_handles);
```

Could be written as:

```rust
let join_handles: Vec<JoinHandle<T>> = ...;
for handle in join_handles {
    handle.await;
}
```
Note that the `handle.await` will trigger a `#[must_use]` warning in the above, which is not triggered in the original example.
